src/preprocess.js

```diff
@@ -95,21 +101,27 @@ async function replaceGeometryIndexes() {
 }
 
 function updateEncodingInner(filename) {
-  return new Promise((resolve, reject) => {
+  return new Promise((resolve) => {
     const filePath = path.join(__dirname, "..", "data", filename);
-    const tempPath = `${filePath}.tmp`;
+    const destPath = path.join(__dirname, "..", "processed", filename);
+
     const inStream = fs.createReadStream(filePath);
-    const outStream = fs.createWriteStream(tempPath);
-    inStream.pipe(iconv.decodeStream("ISO-8859-1")).pipe(outStream);
+    const outStream = fs.createWriteStream(destPath);
+
+    inStream
+      .pipe(iconv.decodeStream("ISO-8859-1"))
+      .pipe(iconv.encodeStream("utf8"))
+      .pipe(outStream);
+
     outStream.on("close", () => {
-      fs.rename(tempPath, filePath, error => (error ? reject(error) : resolve()));
+      resolve();
     });
   });
 }
 
 async function updateEncoding() {
-  for (let i = 0; i < filenames.length; i += 1) {
-    await updateEncodingInner(filenames[i]);
+  for (const file of filenames) {
```

Comment:
Previously this loop didn't actually wait for the encoding to finish. A for/of loop waits for promises.

Reply:
Interesting. Did you test that this change has an effect? for and for/of should work the same - it's forEach that's different because each iteration of the loop is a function call there. Here, await should be doing the waiting, not for/of.

Reply:
Damn, you are right! I was under the very strong impression that only for/of loops can await promises, but after a quick googling it seems that all native loops can. I was wondering how this could have worked previously :smile: 

Anyway, a for/of loop is nice and works as well.